util/src/main/java/org/datacommons/util/StatChecker.java

```diff
@@ -98,12 +97,21 @@ public StatChecker(LogWrapper logCtx, Set<String> samplePlaces, boolean verbose)
     }
   }
 
-  // Given a graph, extract time series info (about the chosen sample places) from the
-  // statVarObservation nodes and save it into seriesSummaryMap.
-  public void extractSeriesInfoFromGraph(McfGraph graph) {
+  // Given a graph, extract stat var info and time series info (about the chosen sample places) from
+  // the statVarObservation nodes.
+  public void extractStatsFromGraph(McfGraph graph) {
     for (Map.Entry<String, McfGraph.PropertyValues> node : graph.getNodesMap().entrySet()) {
-      if (isSvObWithNumberValue(node.getValue()) && shouldExtractSeriesInfo(node.getValue())) {
-        extractSeriesInfoFromNode(node.getValue());
+      if (isSvObWithNumberValue(node.getValue())) {
+        // We will extract basic stat var information from every StatVarObservation nodes
+        extractStatVarInfoFromNode(node.getValue());
+        // We will only extract series information from StatVarObservation nodes about sample places
+        if (shouldExtractSeriesInfo(node.getValue())) {
+          String placeDcid = McfUtil.getPropVal(node.getValue(), Vocabulary.OBSERVATION_ABOUT);
+          PlaceSeriesSummary placeSeriesSummary =
+              placeSeriesSummaryMap.getOrDefault(placeDcid, new PlaceSeriesSummary());
+          placeSeriesSummary.extractSeriesFromNode(node.getValue());
+          placeSeriesSummaryMap.put(placeDcid, placeSeriesSummary);
```

Comment:
Does this need synchronized access? Given the change to do some work in every call, I just wonder if we should make this function itself synchronized now.

Reply:
Actually, yes, I'll make this function itself synchronized and add a todo to look into optimizing this